docs/releasing.md

```diff
@@ -0,0 +1,13 @@
+# Releasing new versions
+
+This checklist is designed to help @robrix remember the steps involved in making a new release.
```

Comment:
I can never remember how to do this, especially the Hackage steps, so I made a checklist.